sormas-ui/src/main/java/de/symeda/sormas/ui/samples/SampleController.java

```diff
@@ -135,6 +137,23 @@ public void onDiscard() {
 		VaadinUiUtil.showModalPopupWindow(createView, I18nProperties.getString(Strings.headingReferSample));
 	}
 
+	private void fillPathogenFields(SampleCreateForm createForm, SampleDto dto) {
```

Comment:
Rename to "fillPathogenTest" and add a PathogenTestDto as a parameter (since the SampleDto won't have the PathogenTestDto as a field itself); the current method name indicates that the fields in the UI are filled while it's actually the other way around.

Reply:
i've refactored a bit the whole save sample part - this method is now integrated there